middlewares/datadog.go

```diff
@@ -0,0 +1,92 @@
+package middlewares
+
+import (
+	"time"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/safe"
+	"github.com/containous/traefik/types"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/dogstatsd"
+)
+
+var _ Metrics = (Metrics)(nil)
```

Comment:
Does this statement have any effect? I saw these kind of constructs to verify a concrete implementation is implementing an interface. To have the `Metrics` on the right hand side doesn't make any sense to me. Can you explain it or clean it up?